lib/ansible/modules/identity/keycloak/keycloak_realm.py

```diff
@@ -0,0 +1,1062 @@
+    realm = module.params.get('name') if module.params.get('name') is not None else module.params.get('realm')
+
+    # convert module parameters to realm representation parameters (if they belong in there)
+    realm_params = [x for x in module.params
+                    if x not in list(keycloak_argument_spec().keys()) + ['state', 'name'] and
+                    module.params.get(x) is not None]
+
+    # If the I(realm) is not set, assume I(name) as the default.
+    if module.params.get('realm') is None:
+        realm_params['realm'] = realm
+
+    before_realm = kc.get_realm_by_name(realm)
+    before_realm = {} if before_realm is None else before_realm
+
+    # Build a proposed changeset from parameters given to this module
+    changeset = dict()
+
+    for realm_param in realm_params:
+        # lists in the Keycloak API are sorted
+        new_param_value = module.params.get(realm_param)
+        if isinstance(new_param_value, list):
+            try:
+                new_param_value = sorted(new_param_value)
+            except TypeError:
+                pass
+        changeset[camel(realm_param)] = new_param_value
+
+    # Whether creating or updating a client, take the before-state and merge the changeset into it
+    updated_realm = before_realm.copy()
+    updated_realm.update(changeset)
+
+    result['proposed'] = changeset
+    result['existing'] = before_realm
+
+    # If the client does not exist yet, before_client is still empty
+    if before_realm == dict():
+        if state == 'absent':
+            # do nothing and exit
+            if module._diff:
+                result['diff'] = dict(before='', after='')
+            result['msg'] = 'Realm does not exist, doing nothing.'
+            module.exit_json(**result)
+
+        # create new realm
+        result['changed'] = True
+
+        # when creating a realm, the id should be the same as the realm name -- otherwise bugs magically appear
+        if 'id' not in updated_realm:
+            updated_realm['id'] = updated_realm['realm']
+
+        if module.check_mode:
+            if module._diff:
+                result['diff'] = dict(before='', after=updated_realm)
+
+            module.exit_json(**result)
+
+        kc.create_realm(updated_realm)
+        after_realm = kc.get_realm_by_name(realm)
+
+        if module._diff:
+            result['diff'] = dict(before='', after=after_realm)
+
+        result['end_state'] = after_realm
+
+        result['msg'] = 'Realm %s has been created.' % realm
+        module.exit_json(**result)
+    else:
+        if state == 'present':
+            # update existing realm
+            result['changed'] = True
+            if module.check_mode:
+                # We can only compare the current client with the proposed updates we have
+                if module._diff:
+                    result['diff'] = dict(before=before_realm,
+                                          after=updated_realm)
+
```

Comment:
Little :bug: : In *check* mode, the `keycloak_realm` always returns `result['changed'] = True`

You need to add something like `result['changed'] = (before_realm != updated_realm)` here...

Reply:
Note that the same addition is also needed in the [keycloak_client](https://github.com/ansible/ansible/blob/6264a55cdc1ef8672b5184376d47621e5b394411/lib/ansible/modules/identity/keycloak/keycloak_client.py#L718) module.

Reply:
Thank you for the review. Your suggested changes are now in the PR, and I opened #37776 for keycloak_client as well.